pkg/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,36 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/consts"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/orchestrator/kubernetes"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/orchestrator/openshift"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/vars"
+)
+
+//go:generate ../../bin/mockgen -destination mock/mock_orchestrator.go -source orchestrator.go
+type Interface interface {
```

Comment:
should we have a Name() method to identify the implementation of the orchestrator ?

also add docstring to the interface